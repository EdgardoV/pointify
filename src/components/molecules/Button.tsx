import React from 'react';
import {TouchableOpacity, StyleSheet, ViewStyle} from 'react-native';
import {TextContainer} from './TextContainer';
import Theme from 'theme';

interface Props {
  label: string;
  style?: ViewStyle;
  onPress: () => void;
}

const Button: React.FC<Props> = ({label, style, onPress}: Props) => {
  const {container} = styles;

  return (
    <TouchableOpacity
      testID="button"
      activeOpacity={0.8}
      style={[container, style]}
      onPress={onPress}>
      <TextContainer
        text={label}
        textAlign="center"
        textColor={Theme.Colors.White}
        fontWeight="Bold"
        fontSize={12}
      />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Theme.Colors.BrightGreek,
    height: 50,
    borderRadius: 10,
    justifyContent: 'center',
  },
});

export {Button};
