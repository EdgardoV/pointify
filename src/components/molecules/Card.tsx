import React from 'react';
import {View, Text, StyleSheet} from 'react-native';
import Theme from 'theme';
import {formatNumber} from 'utils/functions';

interface Props {
  title: string;
  points: number;
}

const Card: React.FC<Props> = ({title, points}: Props) => {
  const {container, label, labelTitle, labelPoints} = styles;
  return (
    <View style={container}>
      <Text style={[label, labelTitle]}>{title}</Text>
      <Text style={[label, labelPoints]}>{`${formatNumber(points)} pts`}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 143,
    backgroundColor: Theme.Colors.BrightGreek,
    marginHorizontal: 33,
    borderRadius: 20,
    shadowColor: Theme.Colors.Black,
    shadowOffset: {width: 0, height: 4},
    shadowOpacity: 0.5,
    shadowRadius: 4,
    justifyContent: 'center',
    alignItems: 'center',
  },
  label: {
    fontFamily: 'Avenir-Black',
    color: Theme.Colors.White,
    textAlign: 'center',
  },
  labelTitle: {
    position: 'absolute',
    top: 21,
    left: 19,
    textTransform: 'capitalize',
  },
  labelPoints: {
    fontSize: 32,
  },
});

export {Card};
