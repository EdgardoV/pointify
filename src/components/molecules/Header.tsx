import React from 'react';
import {View} from 'react-native';
import {TextContainer} from './TextContainer';
import {useTranslation} from 'react-i18next';
import {useColorScheme} from 'hooks/useColorScheme';
import Theme from 'theme';

interface Props {
  title?: string;
}

const Header: React.FC<Props> = ({title}: Props) => {
  const {t} = useTranslation();
  const {colorScheme} = useColorScheme();

  return (
    <View>
      <TextContainer
        text={title || t('home:welcome')}
        fontWeight="Bold"
        fontSize={16}
        textColor={Theme.Colors.PrimaryText[colorScheme]}
      />
      <TextContainer
        text="Edgardo Victorino MarÃ­n"
        fontWeight="Regular"
        fontSize={14}
        textColor={Theme.Colors.PrimaryText[colorScheme]}
      />
    </View>
  );
};

export {Header};
