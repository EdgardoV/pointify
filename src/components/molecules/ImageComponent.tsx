import {PLACEHOLDER_1} from 'assets/images';
import React, {ReactElement, useState} from 'react';
import {Image, ImageStyle, StyleSheet, View} from 'react-native';

interface Props {
  testID?: string;
  width?: number;
  height?: number;
  borderRadius?: number;
  placeholder?: ReactElement;
  style?: ImageStyle;
  uri: string;
  callback?: (data: boolean) => void;
}

const ImageComponent: React.FC<Props> = ({
  width = 55,
  height = 55,
  borderRadius = 10,
  placeholder,
  style,
  uri,
  callback,
}: Props) => {
  const [isImageLoadingError, setIsImageLoadingError] =
    useState<boolean>(false);
  const {container} = styles;

  const handleImageError = () => {
    setIsImageLoadingError(true);
    if (callback !== undefined) {
      callback(true);
    }
  };

  return (
    <View>
      {isImageLoadingError ? (
        <Image
          testID="product-image"
          source={placeholder || PLACEHOLDER_1}
          style={style || [container, {width, height}, {borderRadius}]}
          resizeMode="contain"
        />
      ) : (
        <Image
          testID="product-image"
          source={{uri}}
          style={style || [container, {width, height}, {borderRadius}]}
          onError={handleImageError}
          resizeMode="cover"
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    alignSelf: 'center',
  },
});

export {ImageComponent};
