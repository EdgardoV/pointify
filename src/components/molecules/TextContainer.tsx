import React from 'react';
import {View} from 'react-native';
import {CustomText as Text, CustomTextProps} from '../atoms/CustomText';

export interface TextContainerProps extends CustomTextProps {
  marginRight?: number;
  marginLeft?: number;
  marginTop?: number;
  marginBottom?: number;
  marginHorizontal?: number;
  marginVertical?: number;
}

const TextContainer: React.FC<TextContainerProps> = ({
  marginBottom,
  marginTop,
  marginLeft,
  marginRight,
  marginHorizontal,
  marginVertical,
  text,
  fontSize,
  typography,
  textColor,
  textAlign,
  fontWeight,
  numberOfLines,
  transform,
  underline,
  allowFontScaling,
  spacing,
  lineHeight,
}: TextContainerProps) => (
  <View
    style={{
      marginLeft,
      marginRight,
      marginTop,
      marginBottom,
      marginHorizontal,
      marginVertical,
    }}>
    <Text
      text={text}
      fontSize={fontSize}
      typography={typography}
      textColor={textColor}
      textAlign={textAlign}
      fontWeight={fontWeight}
      numberOfLines={numberOfLines}
      transform={transform}
      underline={underline}
      allowFontScaling={allowFontScaling}
      spacing={spacing}
      lineHeight={lineHeight}
    />
  </View>
);

export {TextContainer};
