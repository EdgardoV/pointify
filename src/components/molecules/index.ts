export * from './Button';
export * from './Card';
export * from './Header';
export * from './ImageComponent';
export * from './ProductTile';
export * from './TextContainer';
