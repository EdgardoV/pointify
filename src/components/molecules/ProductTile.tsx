import React from 'react';
import {StyleSheet, TouchableOpacity, View} from 'react-native';
import {ProductType} from 'reactRedux/types';
import {ImageComponent} from './ImageComponent';
import {TextContainer} from './TextContainer';
import Theme from 'theme';
import {useColorScheme} from 'hooks/useColorScheme';
import {formatDate, formatNumber} from 'utils';
import {TextSpan} from 'components/atoms/TextSpan';
import IconArrow from 'assets/svg/icon-arrow';

interface Props {
  product: ProductType;
  onPress: (product: ProductType) => void;
}

const ProductTile: React.FC<Props> = ({product, onPress}: Props) => {
  const {colorScheme} = useColorScheme();
  const {container, data, points, arrow} = styles;

  return (
    <TouchableOpacity onPress={() => onPress(product)}>
      <View testID="product-container" style={container}>
        <View>
          <ImageComponent testID="product-image" uri={product.image} />
        </View>
        <View style={data}>
          <TextContainer
            text={product.product}
            textColor={Theme.Colors.PrimaryText[colorScheme]}
            fontWeight="Bold"
            fontSize={14}
          />
          <TextContainer
            text={formatDate(product.createdAt)}
            textColor={Theme.Colors.PrimaryText[colorScheme]}
            fontWeight="Regular"
            fontSize={12}
          />
        </View>
        <View style={points}>
          <TextSpan>
            <TextSpan
              text={product.isRedemption ? '-' : '+'}
              textColor={
                product.isRedemption
                  ? Theme.Colors.Red
                  : Theme.Colors.ToxicSludge
              }
              fontWeight="Bold"
              fontSize={16}
            />
            <TextSpan
              text={formatNumber(product.points)}
              textColor={Theme.Colors.PrimaryText[colorScheme]}
              fontWeight="Bold"
              marginLeft={12}
            />
          </TextSpan>
        </View>
        <View style={arrow}>
          <IconArrow />
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    marginBottom: 8,
  },
  data: {
    flex: 2,
    justifyContent: 'center',
    marginHorizontal: 16,
  },
  points: {
    flex: 1,
    justifyContent: 'center',
  },
  arrow: {
    justifyContent: 'center',
  },
});

export {ProductTile};
