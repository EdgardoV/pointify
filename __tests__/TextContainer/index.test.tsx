import React from 'react';
import renderer from 'react-test-renderer';
import {TextContainer} from 'components/molecules/TextContainer';

describe('TextContainer component', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(<TextContainer text="Hello World" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with custom props', () => {
    const tree = renderer
      .create(
        <TextContainer
          text="Hello World"
          fontWeight="Bold"
          fontSize={16}
          textColor="red"
          marginLeft={10}
          marginTop={20}
          marginRight={5}
          marginBottom={15}
          textAlign="center"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
