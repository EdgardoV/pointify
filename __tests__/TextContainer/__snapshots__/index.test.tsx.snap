// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextContainer component renders correctly with custom props 1`] = `
<View
  style={
    {
      "marginBottom": 15,
      "marginHorizontal": undefined,
      "marginLeft": 10,
      "marginRight": 5,
      "marginTop": 20,
      "marginVertical": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    style={
      [
        [
          {
            "color": "#020202",
            "fontFamily": "Avenir",
            "fontSize": 12,
          },
          {
            "fontSize": 16,
          },
          {
            "color": "red",
          },
          {
            "fontFamily": "Avenir-Black",
          },
        ],
        {
          "letterSpacing": undefined,
          "lineHeight": undefined,
          "textAlign": "center",
          "textDecorationLine": "none",
          "textTransform": undefined,
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;

exports[`TextContainer component renders correctly with default props 1`] = `
<View
  style={
    {
      "marginBottom": undefined,
      "marginHorizontal": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "marginVertical": undefined,
    }
  }
>
  <Text
    ellipsizeMode="tail"
    style={
      [
        [
          {
            "color": "#020202",
            "fontFamily": "Avenir",
            "fontSize": 12,
          },
        ],
        {
          "letterSpacing": undefined,
          "lineHeight": undefined,
          "textAlign": undefined,
          "textDecorationLine": "none",
          "textTransform": undefined,
        },
      ]
    }
  >
    Hello World
  </Text>
</View>
`;
